linux-user/elfload.c

```diff
@@ -3173,6 +3195,15 @@ static void load_elf_image(const char *image_name, int image_fd,
 
     for (i = 0; i < ehdr->e_phnum; i++) {
         struct elf_phdr *eppnt = phdr + i;
+
+        //// --- Begin LibAFL code ---
+        #ifdef TARGET_HEXAGON
+        /* Encountered cases where p_type was PT_NULL
+           but the segment should still be loaded. */
+        if((eppnt->p_type == PT_NULL) && eppnt->p_vaddr) eppnt->p_type = PT_LOAD;
```

Comment:
Potentially something we'd want to upstream (?)

Reply:
This part specifically is quite a hack, since it essentially breaks the elf spec. Normally, you would expect loader to ignore segments with the type PT_NULL, but my binary at least contains both PT_NULL segments that should be loaded and PT_NULL segments that shouldn't be loaded (due to having zero size). This is likely because my binary is a firmware image rather than a userspace binary, so it gets loaded in a custom way.

TBH I'd be happier with these changes gone, but then anyone who tried to load a similar binary would encounter the same errors that took me quite a while to debug. That's why I'm only making this change (with PT_NULL segments) for Hexagon (guarded with `#ifdef TARGET_HEXAGON`) as there might be binaries on other platforms that this change would break. But I don't believe this should be upstreamed.